### PR TITLE
Fixed fatal error when updating user_data.json with process exist 1

### PR DESCRIPTION
**Pull request by jianqiu**
When creating vm with ephemeral disk, CPI needs to ssh to deployed vm to do some configuration, and this PR is used to fix the ssh connection refused issue when updating user_data.json of bosh agent.

### CODE REVIEW

File: softlayer/vm/fs_agent_env_service.go [hunk 1]
```diff
@@ -52,5 +53,13 @@ func (s fsAgentEnvService) Update(agentEnv AgentEnv) error {
 		return bosherr.WrapError(err, "Marshalling agent env")
 	}
 
-	return s.softlayerFileService.Upload(s.settingsPath, jsonBytes)
+	for i := 0; i < 30; i++ {
```
Comment (maximilien) [inline softlayer/vm/fs_agent_env_service.go]: add the 30 as a constant

File: softlayer/vm/fs_agent_env_service.go [hunk 2]
```diff
@@ -52,5 +53,13 @@ func (s fsAgentEnvService) Update(agentEnv AgentEnv) error {
...
+		s.logger.Debug(s.logTag, "Updating Agent Env: Making attempt #%d", i)
+		err = s.softlayerFileService.Upload(s.settingsPath, jsonBytes)
+		if err == nil {
+			return nil
+		}
+		time.Sleep(10 * time.Second)
```
Comment (maximilien) [inline softlayer/vm/fs_agent_env_service.go]: change the sleep time as a constant

**Comment by maximilien** [comment]
Please also add a description to your PRs. Thanks.

**Comment by jianqiu** [comment]
@maximilien Updated, please review it, thanks!

File: softlayer/vm/fs_agent_env_service.go [hunk 3]
```diff
@@ -2,11 +2,17 @@ package vm
...
 import (
 	"encoding/json"
+	"time"
 
 	bosherr "github.com/cloudfoundry/bosh-utils/errors"
 	boshlog "github.com/cloudfoundry/bosh-utils/logger"
 )
 
+const (
+	maxAttempts = 30
```
Comment (maximilien) [inline softlayer/vm/fs_agent_env_service.go]: Typically constants are named all caps, e.g., `MAX_ATTEMPTS`

File: softlayer/vm/softlayer_vm.go [hunk 4]
```diff
@@ -292,15 +292,21 @@ func (vm SoftLayerVM) DetachDisk(disk bslcdisk.Disk) error {
 // Private methods
 func (vm SoftLayerVM) extractTagsFromVMMetadata(vmMetadata VMMetadata) ([]string, error) {
 	tags := []string{}
+	status := ""
 	for key, value := range vmMetadata {
-		if key == "tags" {
-			stringValue, ok := value.(string)
-			if !ok {
+		if key == "compiling" || key == "job" || key == "index" || key == "deployment" {
```
Comment (maximilien) [inline softlayer/vm/softlayer_vm.go]: Let's separate the PRs guys. Remove this tag change here

**Comment by jianqiu** [comment]
Since the fixes were included in PR https://github.com/maximilien/bosh-softlayer-cpi/pull/99, so closing it.

**Pull request closed without merging**